common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -319,6 +315,39 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V>, Iterable<@K@ObjectM
         return true;
     }
 
+    public boolean containsKey(Object key) {
+        return containsKey(objectToKey(key));
+    }
+
+    @Override
+    public V get(Object key) {
+        return get(objectToKey(key));
+    }
+
+    @Override
+    public V put(@O@ key, V value) {
+        return put(objectToKey(key), value);
+    }
+
+    @Override
+    public V remove(Object key) {
+        return remove(objectToKey(key));
+    }
+
+    @Override
+    public Set<@O@> keySet() {
+        return keySet;
+    }
+
+    @Override
+    public Set<Entry<@O@, V>> entrySet() {
+        return entrySet;
+    }
+
+    private @k@ objectToKey(Object key) {
+        return (@k@) ((@O@) key).@KEY_NUMBER_METHOD@();
```

Comment:
`objectToKey` currently assumes the object will be of the correct type to translate to a native type.  For example in `IntObjectHashMap` we have:

```java
    private int objectToKey(Object key) {
        return (int) ((Integer) key).intValue();
    }
```

However what if the `Object` is not of the correct type?